### PR TITLE
fix: Add logic to determine resource URLs  for flpSandbox based on project type 

### PR DESCRIPTION
**Pull request by kjose90**
https://github.com/SAP/open-ux-tools/issues/2031

- Improve Resource URL Determination for UShell and UI5 Bootstrap Based on Project Type in `templates/common/add/webapp/test/flpSandbox.html` for writers
- Implemented `getBootstrapResourceUrls` function in `fiori-generator-shared/src/common.ts` to determine resource URLs for UShell bootstrap and UI5 bootstrap.
- Updated snapshots in `fiori-freestyle-writer` and `fiori-elements-writer` to reflect the new URL determination logic - where relative URLs are used for Edmx projects and fully qualified URLs are used for CAP projects

### CODE REVIEW

File: packages/fiori-elements-writer/src/index.ts [hunk 1]
```diff
@@ -82,11 +83,19 @@ async function generate<T extends {}>(basePath: string, data: FioriElementsApp<T
     await addOdataService(basePath, feApp.service, fs);
 
     const coercedUI5Version = semVer.coerce(feApp.ui5?.version)!;
+    // Determine if the changes preview should be enabled based on the project type and UI5 version
```
Comment (IainSAP) [inline packages/fiori-elements-writer/src/index.ts:86]: Might be worth moving the templateOptions logic into a new function to keep the code clean.

Comment (kjose90) [reply packages/fiori-elements-writer/src/index.ts]: > Might be worth moving the templateOptions logic into a new function to keep the code clean and have a single condition for project type so its clear which options are for which project type

I moved this logic [here](https://github.com/SAP/open-ux-tools/pull/2132/files#diff-ff4edbeac8e42e9c7ce005e9ba864236b8e5909f154fde419beb25fa1098fa06R156)

**Review by longieirl — APPROVED** [review]
Based on the offline conversations around this topic, I've a better understanding of how this impacts CAP and Fiori UI projects.
Changes look good.
Test coverage is good.
This will introduce changes into tools-suite, as expected.

**Review by IainSAP — COMMENTED** [review]
Small comment

File: packages/fiori-freestyle-writer/src/index.ts [hunk 2]
```diff
@@ -26,16 +27,31 @@ async function generate<T>(basePath: string, data: FreestyleApp<T>, fs?: Editor)
     await initI18n();
     // Clone rather than modifying callers refs
     const ffApp = cloneDeep(data);
+    // Determine if the project type is 'EDMXBackend'.
+    const isEdmxProjectType = ffApp.app.projectType === 'EDMXBackend';
```
[the remark this answers is not in the document]
Comment (kjose90) [reply packages/fiori-freestyle-writer/src/index.ts]: > Usually we would define variable immediately before use.

Yes thanks for that. I moved it closer to the use.

**Review by IainSAP — APPROVED** [review]
Thx @kjose90 

- Updated as expected
- Tests added to cover

**Pull request merged**